src/contexts/legendContext.tsx

```diff
@@ -13,7 +13,7 @@ import type { ActionType, Shape } from '../types';
 interface LegendLabelState {
   id: string;
   label?: string;
-
+  visibility?: boolean;
```

Comment:
Please rename to `isVisible` (everywhere). The value should not be optional.

Reply:
fixed